### PR TITLE
feat: added prettier actions

### PR DESCRIPTION
**Pull request by prisis**

### CODE REVIEW

**Comment by prisis** [comment]
This is just a proposal

File: .prettierignore [hunk 1]
```diff
@@ -4,3 +4,5 @@ doc/
 examples/browser/js/
 CHANGELOG.md
 package-lock.json
+node_modules/
+test/mocha.opts
```
Comment (Shinigami92) [inline .prettierignore:8]: I think prettier as an option for ignore unknown
Maybe we can configure that with the action also

File: .github/workflows/prettier.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        file_pattern: '**/*.*'
+
+    # This step only runs if prettier finds errors causing the previous step to fail
+    # This steps lists the files where errors were found
+    - name: Prettier Output
+      if: ${{ failure() }}
+      shell: bash
+      run: |
+        echo "The following files are not formatted:"
+        echo "${{steps.prettier-run.outputs.prettier_output}}"
```
Comment (Shinigami92) [inline .github/workflows/prettier.yml:31]:
```suggestion
        echo "${{ steps.prettier-run.outputs.prettier_output }}"
```

File: .github/workflows/prettier.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  prettier:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: lint
+      id: prettier-run
+      uses: rutajdash/prettier-cli-action@v1.0.0
```
Comment (Shinigami92) [inline .github/workflows/prettier.yml:19]: I think we can use so, that we are always on v1 latest
```suggestion
      uses: rutajdash/prettier-cli-action@v1
```

File: .prettierignore (hunk 1 above)
Comment (Shinigami92) [inline .prettierignore:8]: Please format this file in asc order and folders at top and files bottom

Comment (Shinigami92) [reply .prettierignore:8]: I also think node_modules is ignored by default :thinking:

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

File: .prettierignore (hunk 1 above)
Comment (prisis) [reply .prettierignore:8]: Not sure, just added it, will not hurt :D

Comment (prisis) [reply .prettierignore:8]: i will check

Comment (prisis) [reply .prettierignore:8]: i will check

File: .github/workflows/prettier.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  prettier:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: lint
+      id: prettier-run
+      uses: rutajdash/prettier-cli-action@v1
```
Comment (Shinigami92) [inline .github/workflows/prettier.yml:19]:
```node
Error: Unable to resolve action `rutajdash/prettier-cli-action@v1`, unable to find version `v1`
```

:cry:

Comment (Shinigami92) [reply .github/workflows/prettier.yml:19]: Don't know why it works for e.g. `actions/checkout@v2` and other actions I'm used to, but this one

File: .prettierignore (hunk 1 above)
Comment (prisis) [reply .prettierignore:8]: On what files should prettier run?

File: .github/workflows/prettier.yml (hunk 4 above)
Comment (prisis) [reply .github/workflows/prettier.yml:19]: Ah, because you need to create a version v1 to use something like this

File: .prettierignore (hunk 1 above)
Comment (Shinigami92) [reply .prettierignore:8]: everything but ignore unknown https://prettier.io/docs/en/cli.html#--ignore-unknown

File: .github/workflows/prettier.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+jobs:
+  prettier:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: lint
```
Comment (Shinigami92) [inline .github/workflows/prettier.yml:17]: This should not be named "lint"

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

File: .github/workflows/prettier.yml (hunk 5 above)
Comment (prisis) [reply .github/workflows/prettier.yml:17]: Than no name?

Comment (Shinigami92) [reply .github/workflows/prettier.yml:17]: or prettify or check-formatting or something like this
it "conflicts" with (es)lint

Comment (Shinigami92) [reply .github/workflows/prettier.yml:17]: image

this just confused me and I thought I'm in the wrong action

**Comment by prisis** [comment]
@Shinigami92 i did make is simpler

**Comment by damienwebdev** [comment]
@prisis I'm not sure I understand the intent of this PR. Would it ever fail? Wouldn't the lint step beforehand fail first?

**Comment by prisis** [comment]
I think this can be closed, after the eslint, has a plugin for prettier, im correct @Shinigami92?

**Comment by Shinigami92** [comment]
> I think this can be closed, after the eslint, has a plugin for prettier, im correct @Shinigami92?

I think yes

**Pull request closed without merging**